Geometry_Engine/Modify/SplitAtPoints.cs

```diff
@@ -212,111 +212,113 @@ public static List<PolyCurve> SplitAtPoints(this PolyCurve curve, List<Point> po
+                        }
 
-                while (j < tmpResult.Count)
-                {
-                    subResultList.Add(tmpResult[j]);
-                    if (i < onCurvePoints.Count)
-                    {
-                        if (tmpResult[j].IEndPoint().IsEqual(onCurvePoints[i]) || (isClosedAndAntiClockwise && tmpResult[j].IStartPoint().IsEqual(onCurvePoints[i])))
+                        Point ePt = crv.IEndPoint();
+                        if (ptsOnSegment.Any(x => x.SquareDistance(ePt) < sqTol)) //If split at endpoint of current segment
                         {
-                            j++;
-                            break;
+                            result.Add(new PolyCurve { Curves = new List<ICurve> { split.Last() } });   //Add last segment to return list
+                            prev = null;    //Set prev to null
                         }
-                        else if (tmpResult[j].IEndPoint().IsEqual(curve.EndPoint()) || (isClosedAndAntiClockwise && tmpResult[j].IEndPoint().IsEqual(curve.StartPoint())))
+                        else    //If not split at end point of segment
                         {
-                            j++;
-                            break;
+                            prev = new PolyCurve { Curves = new List<ICurve> { split.Last() } };    //Set prev to contain the last split segment
                         }
                     }
-                    j++;
+                }
+                else    //If no splitpoints available
+                {
+                    if (prev != null)   //If prev is set
+                        prev.Curves.Add(crv);   //Add full segment to prev
+                    else
+                        prev = new PolyCurve { Curves = new List<ICurve> { crv } }; //If prev is not set, set it with full segment added
                 }
 
-                if (subResultList.Count > 0)
-                    result.Add(new PolyCurve { Curves = subResultList.ToList() });
-
-                i++;
-            }
-
-            if (curve.IsClosed(tolerance) && !(curve.SubParts()[0] is Circle))
-                if (!curve.StartPoint().IsEqual(onCurvePoints[onCurvePoints.Count - 1]))
+                if (k == subParts.Count - 1)    //Last segment
                 {
-                    List<ICurve> subResultList = new List<ICurve>();
-                    foreach (ICurve subCrv in result[result.Count - 1].ISubParts())
-                        subResultList.Add(subCrv);
-                    foreach (ICurve subCrv in result[0].ISubParts())
-                        subResultList.Add(subCrv);
-                    result.RemoveAt(0);
-                    result.RemoveAt(result.Count - 1);
-                    result.Add(new PolyCurve { Curves = subResultList.ToList() });
+                    if (prev != null)   //If prev is set
+                    {
+                        if (curve.IsClosed(tolerance) && result.Count != 0) //If curve is closed
```

Comment:
Am I correct that this line will become funky in case of disjoint `PolyCurve`? Also your algorithm of adding segments to `prev` may return weird stuff for disjoint curves. Maybe worth disallowing the method for anything that is not closed?

Reply:
You might very much be right in that this might do funky stuff for disjointed curves... Have not tested really actually.

With closed I guess you mean continuous curves? If so, maybe. Honestly not sure how well it worked for disjointed curves before.

As for this line, I can change it to something like:
```suggestion
                        if (curve.IStartPoint().SquareDistance(curve.IEndPoint()) < sqTol && result.Count != 0) //If curve is closed
```

Instead, which would then potentially work for disjointed curves (or at least as well as the other parts do...)